stable/oauth2-proxy/templates/_helpers.tpl

```diff
@@ -30,3 +30,18 @@ Create chart name and version as used by the chart label.
 {{- define "oauth2-proxy.chart" -}}
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+Get the most recent version for the apps API group.
+*/}}
+{{- define "oauth2-proxy.appsAPIVersion" -}}
+{{- if .Capabilities.APIVersions.Has "apps/v1" -}}
+apps/v1
+{{- else if .Capabilities.APIVersions.Has "apps/v1beta2" -}}
+apps/v1beta2
+{{- else if .Capabilities.APIVersions.Has "apps/v1beta1" -}}
```

Comment:
Mmmm, this is unlikely to be accepted since there is a strong push to deprecate k8s 1.7

Reply:
helm officially supports current and current -1 k8s versions. So this change will not pass review :(

Reply:
do you always want to use the upstream chart? 
we can add modified version as a dependency for k10

Reply:
I am adding a version of this chart to K10 anyway. It is preferable to use upstream to avoid divergence.